composer/utils/object_store/uc_object_store.py

```diff
@@ -0,0 +1,175 @@
+        try:
+            from databricks.sdk import WorkspaceClient
+        except ImportError as e:
+            raise MissingConditionalImportError('databricks', conda_package='databricks-sdk>=0.8.0,<1.0') from e
+
+        if not 'DATABRICKS_HOST' in os.environ or not 'DATABRICKS_TOKEN' in os.environ:
+            raise ValueError('Environment variables `DATABRICKS_HOST` and `DATABRICKS_TOKEN` '
+                             'must be set to use Databricks Unity Catalog Volumes')
+
+        if not uri.startswith('dbfs:/Volumes'):
+            raise ValueError('Databricks Unity Catalog Volumes paths should start with "dbfs:/Volumes".')
+        self.path = uri.lstrip('dbfs:')
+
+        self.client = WorkspaceClient()
+
+    def _get_object_path(self, object_name: str) -> str:
+        return os.path.join(self.path, object_name)
+
+    def get_uri(self, object_name: str) -> str:
+        """Returns the URI for ``object_name``.
+
+        .. note::
+
+            This function does not check that ``object_name`` is in the object store.
+            It computes the URI statically.
+
+        Args:
+            object_name (str): The object name.
+
+        Returns:
+            str: The URI for ``object_name`` in the object store.
+        """
+        return f'dbfs:{self._get_object_path(object_name)}'
+
+    def upload_object(self,
+                      object_name: str,
+                      filename: str | pathlib.Path,
+                      callback: Callable[[int, int], None] | None = None) -> None:
+        """Upload a file from local to UC volumes.
+
+        Args:
+            object_name (str): Name of the stored object in UC volumes w.r.t. volume root.
+            filename (str | pathlib.Path): Path the the object on disk
+            callback ((int, int) -> None, optional): Unused
+        """
+        # remove unused variable
+        del callback
+
+        with open(filename, 'rb') as f:
+            self.client.files.upload(self._get_object_path(object_name), f)
+
+    def download_object(self,
+                        object_name: str,
+                        filename: str | pathlib.Path,
+                        overwrite: bool = False,
+                        callback: Callable[[int, int], None] | None = None) -> None:
+        """Download the given object from UC Volumes to the specified filename.
+
+        Args:
+            object_name (str): The name of the object to download i.e. path relative to the root of the volume.
+            filename (str | pathlib.Path): The local path where a the file needs to be downloaded.
+            overwrite(bool, optional): Whether to overwrite an existing file at ``filename``, if it exists.
+                (default: ``False``)
+            callback ((int) -> None, optional): Unused
+
+        Raises:
+            FileNotFoundError: If the file was not found in UC volumes.
+            DatabricksError: If there was any other error querying the Databricks UC volumes.
```

Comment:
do we want to retry these errors? If so, we need to reraise as `ObjectStoreTransientError`